### DISCUSSION TITLE
Ip di example-todo-app non esposto in docker environment

### DISCUSSION THREAD

**Discussion by Carnicero90** [lifeisfoo/react-guida-pratica#12] [Q&A]
Ciao, come da titolo: non so se vada necessariamente considerato un errore/bug (forse no), ma `await fastify.listen({ port: 4000 });`, senza esporre l'host, non funziona (chiaramente). Non so se @lifeisfoo ritenga necessario risolverlo a livello di pacchetto, se non è così volendo posso scrivere una microguida qua nelle discussioni.

**Comment by lifeisfoo** [comment]
Credo di aver capito cosa intendi, però se condividi qui il tuo Dockerfile/compose evito di risponderti in modo sbagliato.
Se preferisci puoi rispondere direttamente spiegando nel dettaglio il problema e mostrando il codice della tua soluzione (la guida per gli altri). Poi vediamo in che categoria mettere tutto.
Grazie

**Reply by Carnicero90** [reply]
Ah no grazie mille io per me stesso ho già risolto, non in modo elegantissimo:

 Il "problema" era nelle impostazioni dell'app fastify nel pacchetto example-todo-app, che come host teneva i valori di default (quindi sull'indirizzo di loopback, non esposto).
 
 Mi sono scaricato la tarball dal registry, copiato il file in una cartella a caso (nel volume del progetto dell'immagine node), npm init, npm install fastify, cambiato la riga in cui lanci fastify in `await fastify.listen({ port: 4000, host: '0.0.0.0' });` e `node nomefile.js`.

**Reply by lifeisfoo** [reply]
Avrei fatto lo stesso al posto tuo ma in questo caso non ce n'era bisogno: qui c'è il codice sorgente delle api https://github.com/lifeisfoo/react-guida-pratica/tree/master/api 🙂.

Parlando di Docker, stai usando il server dentro un container esposto sull'host oppure verso altri container usando compose?

Se condividi il tuo codice, e il perché lo stai usando, magari può essere utile anche ad altri che sono nella stessa situazione.

**Reply by Carnicero90** [reply]
Vabbé sto cecato, l'ho pure clonata... Avessi visto, avrei fatto direttamente una pr per aggiungere un paio di opzioni da riga di comando per specificare host e port. Comunque stasera metto in bella il compose e lo condivido.
